cmd/lakectl/cmd/sst.go

```diff
@@ -0,0 +1,180 @@
+func (r *InMemFileInfo) Sys() interface{}   { return nil }
+
+type InMemReadableFile struct {
+	data *bytes.Reader
+}
+
+func (r *InMemReadableFile) ReadAt(p []byte, off int64) (n int, err error) {
+	return r.data.ReadAt(p, off)
+}
+
+func (r *InMemReadableFile) Close() error {
+	return nil
+}
+
+func (r *InMemReadableFile) Stat() (os.FileInfo, error) {
+	return &InMemFileInfo{
+		name: "stdin",
+		size: r.data.Size(),
+	}, nil
+}
+
+func getIterFromFile(filePath string) (committed.ValueIterator, map[string]string, error) {
+	var file sst.ReadableFile
+	var err error
+	// read from stdin (file has to be seekable/stat-able so we have this weird wrapper
+	if filePath == "-" {
+		data, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return nil, nil, err
+		}
+		file = &InMemReadableFile{
+			bytes.NewReader(data),
+		}
+	} else {
+		file, err = os.Open(filePath)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	// read file descriptor
+	reader, err := sst.NewReader(file, sst.ReaderOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	// create an iterator over the whole thing
+	iter, err := reader.NewIter(nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	// wrap it in a Graveler iterator
+	dummyDeref := func() error { return nil }
+	return sstable.NewIterator(iter, dummyDeref), reader.Properties.UserProperties, nil
+}
+
+func writeRangeSSTable(iter committed.ValueIterator, amount int) *Table {
+	rows := make([][]interface{}, 0)
+	for iter.Next() {
+		if amount != -1 && len(rows)+1 > amount {
+			break
+		}
+		v := iter.Value()
+		gv, err := committed.UnmarshalValue(v.Value)
+		if err != nil {
+			panic(err)
+		}
+		ent, err := rocks.ValueToEntry(gv)
+		if err != nil {
+			panic(err)
+		}
+		rows = append(rows, []interface{}{
+			string(v.Key),
+			fmt.Sprintf("%x", gv.Identity),
+			fmt.Sprint(ent.GetSize()),
+			fmt.Sprint(ent.GetETag()),
+			ent.GetLastModified().AsTime().String(),
+			ent.GetAddress(),
+			fmt.Sprint(ent.GetMetadata()),
+		})
+	}
+	return &Table{
+		Headers: []interface{}{"path", "identity", "size", "etag", "last_modified", "address", "metadata"},
+		Rows:    rows,
+	}
+}
+
+func writeMetaRangeSSTable(iter committed.ValueIterator, amount int) *Table {
+	rows := make([][]interface{}, 0)
+	for iter.Next() {
+		if amount != -1 && len(rows)+1 > amount {
+			break
+		}
+		v := iter.Value()
+		gv, err := committed.UnmarshalValue(v.Value)
+		if err != nil {
+			panic(err)
+		}
+		r, err := committed.UnmarshalRange(gv.Data)
+		if err != nil {
+			panic(err)
+		}
+		rows = append(rows, []interface{}{
+			committed.ID(gv.Identity),
+			string(r.MinKey),
+			string(r.MaxKey),
+			fmt.Sprint(r.Count),
+			fmt.Sprint(r.EstimatedSize),
+		})
+	}
+	return &Table{
+		Headers: []interface{}{"range_id", "min_key", "max_key", "count", "estimated_size"},
+		Rows:    rows,
+	}
+}
+
+var sstCatTemplate = `{{ .Table | table }}`
+
+var sstCmd = &cobra.Command{
+	Use:    "cat-sst <sst-file>",
```

Comment:
I'd like to have, at last eventually:
* `cat-sst --file <sst-file>`
* `cat-sst --metarange <metarange_id>`
* `cat-sst --range <range_id>`.

To allow that in future, could this be a flag arg rather than positional?